ethcore/src/ethereum/ethash.rs

```diff
@@ -270,9 +273,14 @@ impl Engine for Arc<Ethash> {
 	/// Apply the block reward on finalisation of the block.
 	/// This assumes that all uncles are valid uncles (i.e. of at least one generation before the current).
 	fn on_close_block(&self, block: &mut ExecutedBlock) -> Result<(), Error> {
-		let reward = self.ethash_params.block_reward;
+		let mut reward = self.ethash_params.block_reward;
 		let fields = block.fields_mut();
 
+		let eras = fields.header.number() / self.ethash_params.ecip1017_era_rounds;
+		for _ in 0..eras {
```

Comment:
I think it's OK:
```
for _ in 0..0 {}
```

is == `for i := 0; i < 0; i++ {}`, so it won't be called as long as `eras` is zero-indexed (which #L294 confirms).